ofnetAgent.go

```diff
@@ -817,7 +827,15 @@ func (self *OfnetAgent) RemoveNetwork(vlanId uint16, vni uint32, Gw string, Vrf
 // AddUplink adds an uplink to the switch
 func (self *OfnetAgent) AddUplink(uplinkPort *PortInfo) error {
 	// Call the datapath
-	return self.datapath.AddUplink(uplinkPort)
+	err := self.datapath.AddUplink(uplinkPort)
+	if err != nil {
+		return err
+	}
+
+	if self.protopath != nil {
+		self.protopath.SetRouterInfo(uplinkPort)
```

Comment:
do we need it to be a fatal error ? The check at this point is only safety check. Netplugin rightly has the required checks and it wont come to this at all.